mopidy_pandora/actor.py

```diff
@@ -42,12 +43,23 @@ def _next_track(self, station_token):
             self.station_token = station_token
             self.tracks = iter(())
 
-        while True:
-            try:
+        try:
+            track = next(self.tracks)
+            # Check if the track is playable
+            if self.backend.api.playable(track):
+                return track
+            else:
+                # Tracks have expired, retrieve fresh playlist from Pandora
+                self.tracks = self._get_playlist(station_token)
                 return next(self.tracks)
-            except StopIteration:
-                self.tracks = (pandora.models.pandora.PlaylistItem.from_json(self.backend.api, station)
-                               for station in self.backend.api.get_playlist(self.station_token)['items'])
```

Comment:
It seemed to me that this try-catch block inside ```while True:``` created the potential for an invite loop. I've changed it so that the PlayBackProvided will only retry once on either (a) the track has expired and is not playable, or (b) we've reached the end of the playlist and need to retrieve a new one.